### ISSUE TITLE
Web component: Filter Panel - Screen reader is not identifying filter options  (Technologies, Business needs )

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#6647]
## Detailed description

Issue:
1. The role i.e. control type is not identified by the screen reader. Hence the user would not know that these are interactive elements
2. The status i.e. whether the options is selected or not is not announced by the screen reader.

Expected:
1. Screen reader is expected to identify the role i.e. control type
2. Screen reader should convey the toggle status whether the option is "selected/checked or unselected/unchecked"

## Additional information

![image.png](https://images.zenhubusercontent.com/5e21a7b470118a498429a20e/7c8505f6-83f5-4e22-ad9e-0a0a0e6ff301)
